### ISSUE TITLE
Squash pull requests instead of merge for cleaner commit history on az-hop

### ISSUE THREAD

**Issue by ltalirz** [Azure/az-hop#1686]
Given that teams like ours need to frequently rebase on the `main` branch of this repository, would it perhaps be possible to switch from using `merge` as the default mode for pull requests to `Squash and merge`? 

This would not cost any extra time, and it would have both the advantage of a cleaner commit history with topical commits, and also reduce the likelihood of having to do extra work in resolving merge conflicts (changing things back and forth). 

cc @matt-chan

**Comment by xpillons** [comment]
Yes, good suggestion. I'm doing this now, so I will close the issue.
